### PR TITLE
Fix SubgraphMatcher for case of no anchor found

### PR DESCRIPTION
**Pull request by SherlockNoMad**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #86430
* __->__ #86421

### CODE REVIEW

File: test/test_fx_passes.py [hunk 1]
```diff
@@ -664,6 +664,25 @@ def pattern(a, a_dtype, a_scale, b, b_dtype, b_scale, out_scale):
         TestCase(False, False, 1),
     ]
 
+class NoAnchorFound:
```
Comment (jerryzh168) [inline test/test_fx_passes.py:667]: can we add some comment talking about what no anchor found means?

**Review by jerryzh168 — APPROVED** [review]
looks good, thanks for the fix. would be better if include at least Summary and Test section in https://github.com/pytorch/pytorch/pull/86421#issue-1400333757 as well

File: test/test_fx_passes.py (hunk 1 above)
Comment (SherlockNoMad) [reply test/test_fx_passes.py:667]: Updated.

**Pull request closed without merging**